ir/irutils.h

```diff
@@ -0,0 +1,98 @@
+#ifndef IR_IRUTILS_H_
+#define IR_IRUTILS_H_
+
+#include <vector>
+
+#include <boost/none.hpp>
+#include <boost/optional/optional.hpp>
+
+#include "frontends/common/constantFolding.h"
+#include "frontends/p4/strengthReduction.h"
+#include "ir/ir.h"
+#include "lib/big_int_util.h"
+#include "lib/cstring.h"
+#include "lib/error.h"
+#include "lib/exceptions.h"
+
+namespace IR {
+
+/// Utility functions for generating IR nodes.
+//
+// Some of these are just thin wrappers around functions in the IR, but it's nice having everything
+// in one place.
+class IRUtils {
```

Comment:
since this class has no instances and no non-static members, maybe it should be a namespace?

or maybe even put all these functions in namespace IR directly.